### PR TITLE
rz-run resolve_value: Pass single-quoted strings unaltered

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [X] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This pr fixes the following error (https://github.com/rizinorg/rizin/actions/runs/5997466915/job/16263926133?pr=3800#step:19:484)

rz-run-ERROR

by making `run.c`'s `resolve_value()` (used by rz-run `setenv` among others) pass single-quoted strings unaltered.

**Test plan**

The change is appropriate. All builds are green.

**Closing issues**

...

### CODE REVIEW

File: test/db/archos/linux-x64/dbg [hunk 1]
```diff
@@ -95,3 +95,13 @@ EXPECT_ERR=<<EOF
 child received signal 9
 EOF
 RUN
+
+NAME=rizin -d no error
+FILE=--
+CMDS=<<EOF
+envDISPLAY=:0
```
Comment (ret2libc) [inline test/db/archos/linux-x64/dbg]: why no space after `env`?

Comment (kazarmy) [reply test/db/archos/linux-x64/dbg]: as per below ... surprisingly it does work

env-help

Comment (kazarmy) [reply test/db/archos/linux-x64/dbg]: hmm maybe i shoudn't depend too much on typos

Comment (kazarmy) [reply test/db/archos/linux-x64/dbg]: 9a93959

Comment (ret2libc) [reply test/db/archos/linux-x64/dbg]: Yeah that's an unexpected behaviour due to the fact `env` was `%` before. Should be fixed, creating an issue now!

**Comment by kazarmy** [comment]
[Header comment](https://github.com/rizinorg/rizin/pull/3800#issue-1869018067) changed

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**